lib/puppet/provider/service/windows.rb

```diff
@@ -73,7 +73,10 @@ def start
       end
     end
 
-    net(:start, @resource[:name])
+    Puppet::Util::Windows::Service.open_service(@resource[:name]) do |service|
```

Comment:
I don't think we should let the service provider do this type of low-level manipulation of service handlers. It would probably be better to have public API methods like:
```
Windows::Service.start(service_name)
Windows::Service.stop(service_name)
Windows::Service.status(service_name)
```

that we can call out to from here.

Reply:
Yeah, I was trying to allow as much flexibility in the util module, but I think you're right about this.